crm_compassion/models/weekly_demand.py

```diff
@@ -136,6 +154,34 @@ def _default_unsponsored_ambassador(self):
             'crm_compassion.number_children_ambassador'))
         return allocate_per_week - (float(ambass_sponsored) / 52.0)
 
+    @api.multi
+    def _compute_resupply_sub(self):
+        """ Compute SUB resupply. """
+        subs = list()
+        average_sub = self._compute_demand_sub()
+        sub_sponsorships = self.env['recurring.contract'].search([
+            ('origin_id.type', '=', 'sub'),
+        ])
+
+        for sub in sub_sponsorships:
+            if sub.end_date and sub.start_date:
```

Comment:
If you add in your search condition that it must have a start date and end date, you won't need this test.